addons/binding/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/handler/HeosGroupHandler.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.heos.handler;
+
+import static org.openhab.binding.heos.HeosBindingConstants.*;
+import static org.openhab.binding.heos.internal.resources.HeosConstants.GID;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.PlayPauseType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.heos.internal.api.HeosFacade;
+import org.openhab.binding.heos.internal.api.HeosSystem;
+import org.openhab.binding.heos.internal.resources.HeosGroup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HeosGroupHandler} handles the actions for a HEOS group.
+ * Channel commands are received and send to the dedicated channels
+ *
+ * @author Johannes Einig - Initial contribution
+ */
+public class HeosGroupHandler extends HeosThingBaseHandler {
+
+    private String gid;
+    private HeosGroup heosGroup;
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
```

Comment:
Please put logger's on top and put a static reference to the class:

```java
    private final Logger logger = LoggerFactory.getLogger(HeosGroupHandler.class);
```

And check all